### PR TITLE
negative index added for selectors (closes #738)

### LINKED ISSUE #738: Implement possibility to pass a negative index value in Selector
If a negative number is given, it is treated as `countElms + index` where `countElms` is the number of the elements, which correspond the selector function.

### PR DESCRIPTION
**Pull request by helen-dikareva**
/cc @AlexanderMoskovkin

### CODE REVIEW

**Comment by helen-dikareva** [comment]
@testcafe-build-bot \retest

File: test/functional/fixtures/api/es-next/selector/testcafe-fixtures/selector-test.js [hunk 1]
```diff
@@ -352,11 +352,11 @@ test('Selector "index" option', async () => {
     expect(el.id).eql('el2');
 
     // Function selector
-    const getThirdEl = Selector(() => document.querySelectorAll('.idxEl'), { index: 2 });
+    const getThirdEl = Selector(() => document.querySelectorAll('.idxEl'), { index: -1 });
```
Comment (AlexanderMoskovkin) [inline test/functional/fixtures/api/es-next/selector/testcafe-fixtures/selector-test.js]: It's `gerLastEl` now?

File: src/errors/runtime/type-assertions.js [hunk 2]
```diff
@@ -2,8 +2,17 @@ import { isFinite, isRegExp, isNil as isNullOrUndefined } from 'lodash';
 import { APIError } from './';
 import MESSAGE from './message';
 
+function isNumber (value) {
+    return isFinite(value) && typeof value === 'number';
+}
+
 function isNonNegativeNumber (value) {
-    return isFinite(value) && value >= 0;
+    return isNumber && value >= 0;
```
Comment (kirovboris) [inline src/errors/runtime/type-assertions.js]: ```return isNumber(value) && value >= 0;``` ?

File: src/errors/runtime/type-assertions.js [hunk 3]
```diff
@@ -2,8 +2,17 @@ import { isFinite, isRegExp, isNil as isNullOrUndefined } from 'lodash';
 import { APIError } from './';
 import MESSAGE from './message';
 
+function isNumber (value) {
```
Comment (kirovboris) [inline src/errors/runtime/type-assertions.js]: Why don't we use ```lodash.isNumber()``` function?

Comment (helen-dikareva) [inline src/errors/runtime/type-assertions.js]: Because we need exactly `isFinite` check https://lodash.com/docs/4.17.2#isNumber.
But we can simplify it

**Comment by helen-dikareva** [comment]
@testcafe-build-bot \retest

**Comment by helen-dikareva** [comment]
@testcafe-build-bot \retest

**Comment by helen-dikareva** [comment]
@testcafe-build-bot \retest

**Comment by helen-dikareva** [comment]
FPR

**Pull request merged**